core/engine_test.go

```diff
@@ -371,6 +371,68 @@ func TestEngine_processSamples(t *testing.T) {
 		assert.IsType(t, &stats.GaugeSink{}, e.Metrics["my_metric"].Sink)
 		assert.IsType(t, &stats.GaugeSink{}, e.Metrics["my_metric{a:1}"].Sink)
 	})
+	t.Run("apply run tags", func(t *testing.T) {
+		tags := map[string]string{"foo": "bar"}
+		e, err, _ := newTestEngine(nil, lib.Options{RunTags: tags})
+		assert.NoError(t, err)
+
+		c := &dummy.Collector{}
+		e.Collector = c
+
+		t.Run("sample untagged", func(t *testing.T) {
+			c.Samples = nil
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			go func() {
+				c.Run(ctx)
+			}()
+
+			for !c.IsRunning() {
+				time.Sleep(time.Millisecond)
```

Comment:
See the notes about the `IsRunning()` method below, this should be refactored. Or possibly removed, I'm not sure if it's even needed and if we actually need to have a context, cancel it, `Run()` the collector and check when it finishes... This is a test and we know it's the `Dummy` collector anyway, so can't we skip all of that?